api/filters_test.go

```diff
@@ -1129,6 +1172,68 @@ func TestRequestForwardingMiddleware(t *testing.T) {
 				So(len(filterFlexAPIMock.ForwardRequestCalls()), ShouldEqual, 0)
 			})
 		})
+
+		Convey("When there is a PUT request to /filter-outputs/test-output-id and the dataset type is cantabular", func() {
+			filterFlexMock, datasetAPIMock, datastoreMock := mock.GenerateMocksForMiddleware(200, cantabularFlexibleTable, 1, "not-applicable")
+
+			filterApi := api.Setup(conf, mux.NewRouter(), datastoreMock, &mock.FilterJob{}, datasetAPIMock, filterFlexMock)
+
+			r, err := http.NewRequest("PUT", "http://localhost:22100/filter-outputs/test-output-id", nil)
+			So(err, ShouldBeNil)
+
+			filterApi.Router.ServeHTTP(w, r)
+			Convey("A call is made to datastore to check the dataset type", func() {
+				So(len(datasetAPIMock.GetCalls()), ShouldEqual, 1)
+			})
+
+			// only need to test once as if it works once, works for all other routes
+			Convey("A call is not made to datastore to check the filter type", func() {
+				So(len(datastoreMock.GetFilterCalls()), ShouldEqual, 0)
+			})
+			Convey("The request is forwarded to dp-cantabular-filter-flex-api", func() {
+				So(len(filterFlexMock.ForwardRequestCalls()), ShouldEqual, 1)
+			})
+		})
+
+		Convey("When there is a GET request to /filters/{filter_blueprint_id}/dimensions and the dataset type is cantabular", func() {
+			filterFlexMock, datasetAPIMock, datastoreMock := mock.GenerateMocksForMiddleware(200, cantabularFlexibleTable, 1, "not-applicable")
```

Comment:
What's this 200? If it's a status code it's better to use the const `http.StatusOK` for clarity